evalml/automl/automl_search.py

```diff
@@ -553,6 +554,8 @@ def _find_best_pipeline(self):
                     X_train, X_threshold_tuning, y_train, y_threshold_tuning = split_data(X_train, y_train, self.problem_type,
                                                                                           test_size=0.2,
                                                                                           random_state=self.random_state)
+                if getattr(self.data_splitter, "transform", None):
```

Comment:
If we used a sampler splitter, we want to transform the data so that we can train the best pipeline, if needed

Reply:
I suggest you use `hasattr` instead